### ISSUE TITLE
[QUESTION] <Support for MAE?>

### ISSUE THREAD

**Issue by YingdiZhang** [lightly-ai/lightly-train#511]
### 🤔 What’s your question?

Thanks for the great job. Do you have the plan to support MAE self-supervised learning? I find simCLR is supported now.

### 🧠 What did you try?

Explain what you’ve already looked into or attempted.

### 📌 Related code or context (if any)

Optional, but helpful for understanding your context.

```python
# Related snippet if relevant
```

**Comment by stegmuel** [comment]
Hi,

It is on our list of potential next features.
Note that it is already implemented in [LightlySSL](https://github.com/lightly-ai/lightly) (corresponding doc is [here](https://docs.lightly.ai/self-supervised-learning/examples/mae.html)).

**Comment by stegmuel** [comment]
As a follow-up question, may I ask why you want MAE and where available pre-training methods (DINOv2, SimCLR, distillation) fell short?

**Comment by YingdiZhang** [comment]
Thanks for your reply. In my area which is infrared image recognition, the foundation models are based on MAE-like models. And I also found DINOv2 pretraining don't work in my area. By the way, If I pretrain a model like MAE using LightlySSL, can I distill it to other model architerctures using lightly-train?

**Comment by stegmuel** [comment]
Thank you for the precision, that's valuable.
As of now we do not support using custom teacher models.
Nonetheless, in earlier experiment we tried using MAE models as teachers in distillation and the results were not that great. A more specific distillation approach like this [one](https://openaccess.thecvf.com/content/CVPR2023/papers/Bai_Masked_Autoencoders_Enable_Efficient_Knowledge_Distillers_CVPR_2023_paper.pdf) might be preferred, but the masking restricts the student architectures.
